lib/ansible/modules/cloud/amazon/aws_s3.py

```diff
@@ -714,6 +748,21 @@ def main():
         else:
             module.fail_json(msg="Bucket parameter is required.")
 
+    # Copy an object from a bucket, not the entire bucket
+    if mode == 'copyobj':
+        if obj is None:
+            module.fail_json(msg="object parameter is required")
```

Comment:
Rather than doing validation like this here, you should be able to add a `required_if` option to AnsibleModule:
```
    module = AnsibleModule(
        argument_spec=argument_spec,
        supports_check_mode=True,
        required_if=[('mode', 'copyobj', ('obj',))]
    )
```